src/main/java/com/exactpro/th2/conn/ConnectivityBatch.java

```diff
@@ -81,21 +81,36 @@ private static void checkMessages(List<ConnectivityMessage> iMessages, String se
         if (!iMessages.stream()
                 .allMatch(iMessage -> Objects.equals(sessionAlias, iMessage.getSessionAlias())
                         && direction == iMessage.getDirection())) {
-            throw new IllegalArgumentException("List " + iMessages + " has elemnts with incorrect metadata, expected session alias '"+ sessionAlias +"' direction '" + direction + '\'');
+            throw new IllegalArgumentException("List " + iMessages + " has elements with incorrect metadata, expected session alias '"+ sessionAlias +"' direction '" + direction + '\'');
         }
 
-        if (!IntStream.range(0, iMessages.size() - 1)
-                .allMatch(index -> iMessages.get(index).getSequence() + 1 == iMessages.get(index + 1).getSequence())) {
-            if (LOGGER.isErrorEnabled()) {
-                LOGGER.error("List {} hasn't elements with incremental sequence with one step", iMessages.stream()
-                                            .map(ConnectivityMessage::getSequence)
-                                            .collect(Collectors.toList()));
+        boolean sequencesUnordered = false;
+        List<Long> missedSequences = new ArrayList<>();
+        for (int index = 0; index < iMessages.size() - 1; index++) {
+            long nextExpectedSequence = iMessages.get(index).getSequence() + 1;
+            long nextSequence = iMessages.get(index + 1).getSequence();
+            if (nextExpectedSequence != nextSequence) {
+                sequencesUnordered = true;
             }
+            LongStream.range(nextExpectedSequence, nextSequence).forEach(missedSequences::add);
+        }
```

Comment:
Maybe move the current implementation into if (LOGGER.isErrorEnabled()

Reply:
done